Index: device/lib/z80/Makefile.in
===================================================================
--- device/lib/z80/Makefile.in	(Revision 10047)
+++ device/lib/z80/Makefile.in	(Arbeitskopie)
@@ -6,7 +6,7 @@
 
 LIB_TYPE     = @LIB_TYPE@
 
-SCC = $(top_builddir)/bin/sdcc -mz80
+SCC = $(top_builddir)/bin/sdcc -mz80 --dump-graphs
 SAS = $(top_builddir)/bin/sdasz80
 
 # override PORTDIR defined by super (parent) makefile
