Index: device/lib/gbz80/Makefile.in
===================================================================
--- device/lib/gbz80/Makefile.in	(Revision 10088)
+++ device/lib/gbz80/Makefile.in	(Arbeitskopie)
@@ -6,7 +6,7 @@
 
 LIB_TYPE     = @LIB_TYPE@
 
-SCC = $(top_builddir)/bin/sdcc -mgbz80
+SCC = $(top_builddir)/bin/sdcc -mgbz80  --nogcse --dump-graphs
 SAS = $(top_builddir)/bin/sdasgb
 
 # override PORTDIR defined by super (parent) makefile
